src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -474,6 +478,28 @@ private DType(DKind kind, string namedValueKind)
             AssertValid();
         }
 
+        internal DType(LazyTypeProvider provider, bool isTable)
+        {
+            Contracts.AssertValue(provider);
+            
+            LazyTypeProvider = provider;
+            Kind = isTable ? DKind.LazyTable : DKind.LazyRecord;
```

Comment:
>isTable ? DKind.LazyTable : DKind.LazyRecord [](http://example.com/codeflow?start=19&length=44)

Should this be a property on type provider?
Kind = provider.TypeKind

Reply:
No, reusing the same provider for records/tables of the same object is useful, so we don't need a new provider instance for something like `First(LazyTableType)`